### PR TITLE
💥 add alexchomiak/goto-react-app-2.0 boilerplate

### PR DESCRIPTION
**Pull request by bmiddha**
- ✨ Migrated to alexchomiak/goto-react-app-2.0
- 🧪 Updated front-end testing pipeline.

### CODE REVIEW

File: client/config/environment.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+ *
+ * Author Github: https://github.com/alexchomiak
+ */
+
+// ? Environment constants in project (API Keys, etc)
+// * If storing any important data, make sure to add to your git ignore!
+
+module.exports = {
+    ENV_VARIABLE_EXAMPLE: 'ENVIRONMENT VARIABLE!! :)'
+}
```
Comment (anarmawala) [inline client/config/environment.js:17]: Could be used for Development Docker Environment! 👍

File: client/config/settings.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        // ? Base directory is src for aliases
+        '@utils': ['utils'],
+        '@actions': ['models/redux/actions'],
+        '@reducers': ['models/redux/reducers'],
+        '@redux': ['models/redux'],
+        '@components': ['components'],
+        '@styles': ['styles'],
+        '@assets': ['assets'],
+        '@interfaces': ['models/interfaces'],
+        '@pages': ['pages']
```
Comment (anarmawala) [inline client/config/settings.js:36]: This can be used for imports in current files! 🎇

File: client/src/test/NavBar.test.tsx [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+        </Provider>
+    )
+
+    // * Test for Links in App
+    expect(screen.getByText('Registrum')).toBeDefined()
+    expect(screen.getByText('About Us')).toBeDefined()
+    expect(screen.getByText('Contact')).toBeDefined()
+    expect(screen.getByText('Login')).toBeDefined()
+    expect(screen.getByText('Register')).toBeDefined()
+})
```
Comment (anarmawala) [inline client/src/test/NavBar.test.tsx:22]: Can we implement authenticated NavBar tests?

**Review by anarmawala — APPROVED** [review]

File: client/src/test/NavBar.test.tsx (hunk 3 above)
Comment (bmiddha) [reply client/src/test/NavBar.test.tsx:22]: It's possible now. Tests can be added later.

Comment (alexchomiak) [reply client/src/test/NavBar.test.tsx:22]: Implementing a login workflow in clientside tests creates a bunch of problems we would need to work around. We need to figure out how to create a login flow for each test that needs auth, and how do we communicate with an API in all test environments? i.e. GitHub actions, various cloud providers

**Pull request merged**